src/controller/AutoCommissioner.cpp

```diff
@@ -133,7 +133,20 @@ void AutoCommissioner::StartCommissioning(CommissioneeDeviceProxy * proxy)
 {
     // TODO: check that there is no commissioning in progress currently.
     mCommissioneeDeviceProxy = proxy;
-    mCommissioner->PerformCommissioningStep(mCommissioneeDeviceProxy, CommissioningStage::kArmFailsafe, mParams, this);
+    mCommissioner->PerformCommissioningStep(mCommissioneeDeviceProxy, CommissioningStage::kArmFailsafe, mParams, this, 0,
+                                            GetCommandTimeout(CommissioningStage::kArmFailsafe));
+}
+
+Optional<System::Clock::Timeout> AutoCommissioner::GetCommandTimeout(CommissioningStage stage)
+{
+    switch (stage)
+    {
+    case CommissioningStage::kWiFiNetworkEnable:
+    case CommissioningStage::kThreadNetworkEnable:
+        return Optional<System::Clock::Timeout>(System::Clock::Timeout(System::Clock::Seconds16(30)));
+    default:
+        return Optional<System::Clock::Timeout>();
```

Comment:
Would be good to insert comment that this eventually results in using the default timeout specified in the IM.

Side-question: Do you think the commissioner should be more prescriptive about specifying timeouts for all commands it uses, and not rely on the IM default (which can change...)?

Reply:
I think most of the commands are expected to complete quite rapidly, so the default feels appropriate. I'm not sure that setting them explicitly adds anything currently.

Reply:
Added a comment